superset-frontend/src/explore/components/SaveModal.tsx

```diff
@@ -174,15 +174,13 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
       if (this.props.datasource?.type === DatasourceType.Query) {
         const { schema, sql, database } = this.props.datasource;
         const { templateParams } = this.props.datasource;
-        const columns = this.props.datasource?.columns || [];
 
         await this.props.actions.saveDataset({
           schema,
           sql,
           database,
           templateParams,
           datasourceName: this.state.datasetName,
-          columns,
```

Comment:
why do we no longer pass in the columns when saving?

Reply:
if you look inside the `create` command for `Datasets` the function `fetch_metdata` actually handles it based upon the sql/query defined in the payload:

https://github.com/preset-io/superset/blob/d86647916f5a72df1f2f8b83fb11d0c5bc54ae71/superset/connectors/sqla/models.py#L1258

Reply:
in the original `superset/sqllab_viz` endpoint we had to build out these object ourselves so intially that I needed to add the logic but realize soon after it was builtin